framework/scenario/src/scenario/model/transaction/tx_response.rs

```diff
@@ -80,39 +58,100 @@ impl TxResponse {
         }
     }
 
+    /// Creates a scenario "expect" field based on the real response.
+    ///
+    /// Useful for creating traces that also check the results come out always the same.
+    pub fn to_expect(&self) -> TxExpect {
+        if self.tx_error.is_success() {
+            let mut tx_expect = TxExpect::ok();
+            if self.out.is_empty() {
+                tx_expect = tx_expect.no_result();
+            } else {
+                for raw_result in &self.out {
+                    let result_hex_string = format!("0x{}", hex::encode(raw_result));
+                    tx_expect = tx_expect.result(result_hex_string.as_str());
+                }
+            }
+            tx_expect
+        } else {
+            TxExpect::err(
+                self.tx_error.status,
+                format!("str:{}", self.tx_error.message),
+            )
+        }
+    }
+
+    fn process_signal_error(&self) -> TxResponseStatus {
+        if let Some(event) = self.find_log(LOG_IDENTIFIER_SIGNAL_ERROR) {
+            let topics = event.topics.as_ref();
+            if let Some(error) = process_topics_error(topics) {
+                return TxResponseStatus::signal_error(&error);
+            }
+
+            let error_raw = base64::decode(topics.unwrap().get(1).unwrap()).unwrap();
+            let error = String::from_utf8(error_raw).unwrap();
+            return TxResponseStatus::signal_error(&error);
+        }
+
+        TxResponseStatus::default()
+    }
+
     fn process(self) -> Self {
-        self.process_out().process_new_deployed_address()
+        self.process_out()
+            .process_new_deployed_address()
+            .process_new_issued_token_identifier()
     }
 
     fn process_out(mut self) -> Self {
-        if let Some(first_scr) = self.api_scrs.get(0) {
-            self.out = decode_scr_data_or_panic(first_scr.data.as_str());
+        let out_scr = self
+            .api_scrs
+            .iter()
+            .find(|scr| scr.nonce != 0 && scr.data.starts_with('@'));
```

Comment:
Lambda function can be extracted as a separate, private function, e.g. `containsReturnData`.

Reply:
This was my initial ideea, done!